Source/WebInspectorUI/UserInterface/Models/DOMNodeStyles.js

```diff
@@ -459,6 +459,37 @@ WI.DOMNodeStyles = class DOMNodeStyles extends WI.Object
         return result.promise;
     }
 
+    changeGroupingText(grouping, text)
```

Comment:
i think you can make this cleaner using `async`
```
async changeGroupingText(grouping, text)
{
    text ||= "";
    
    this._needsRefresh = true;
    this.ignoreNextContentDidChangeForStyleSheet = grouping.ownerStyleSheet;

    let target = WI.assumingMainTarget();
    let {grouping: groupingPayload} = await target.setGroupingHeaderText(grouping.id, text);
    
    target.DOMAgent.markUndoableState();
    await this.refresh();

    return groupingPayload;
}
```

also, why not have this be on `WI.CSSGrouping` instead of `WI.DOMNodeStyles`?  you're passing the `WI.CSSGrouping` into this method anyways, so it seems like that's really the focus here, not the `WI.DOMNodeStyles` (which is really only used for the `refresh`)